types/node/v18/perf_hooks.d.ts

```diff
@@ -454,6 +472,104 @@ declare module "perf_hooks" {
                 },
         ): void;
     }
+    /**
+     * Provides detailed network timing data regarding the loading of an application's resources.
+     *
+     * The constructor of this class is not exposed to users directly.
+     * @since v18.2.0, v16.17.0
+     */
+    class PerformanceResourceTiming extends PerformanceEntry {
+        protected constructor();
```

Comment:
```suggestion
    class PerformanceResourceTiming extends PerformanceEntry {
        protected constructor();
        readonly entryType: "resource";
```
The other PerformanceEntry subclasses narrow their `entryType` property, so may as well add this here, and to the existing PerformanceResourceTiming class definitions in v18/v20/v22. (`"resource"` will need adding to the `EntryType` union as well.)

Reply:
Good catch! I also created https://github.com/nodejs/node/issues/54212 because "resource" not documented currently.

Reply:
I have added two new entry types: `'resource'` and `'net'`.